### PR TITLE
fix wrong TVL on pool tab

### LINKED ISSUE #1208: Pool value representation does not add up
Let's look at this pool which is being looked at with the account who created the pool:

`did:op:d855538c659038bedbdb094417391ac63a9d22084900cf510bb78ed66f2ac6da`
https://rinkeby.etherscan.io/address/0x7807e6ac970fd11cac8e491b711b70c5187aa9ee
Screen Shot 2022-03-17 at 14 09 34

It was created by adding 50 OCEAN from user wallet:
https://rinkeby.etherscan.io/tx/0x9465e0f040323faedfa94ea0e938cab382c9a9ea60c7ba831ef2e868c6c3ee5e

But numbers suggest the total pool TVL doubled. No further transactions happened on that pool.

If I get this right, user value locked & owner value locked is correct, but total TVL should be simply half, so also 50.

### PR DESCRIPTION
**Pull request by EnzoVezzaro**
- Fixes #1208 

Changes proposed in this PR:

- remove multiplier from totalLiquidityInOcean

### CODE REVIEW

**Comment by kremalicious** [comment]
What's up with the OCEAN/DT numbers at bottom? Not sure we'll keep this there but now it's confusing again as this shows the actual OCEAN/DT values, so in this case the OCEAN there should be half:

Screen Shot 2022-03-30 at 10 30 28

**Comment by EnzoVezzaro** [comment]
> What's up with the OCEAN/DT numbers at bottom? Not sure we'll keep this there but now it's confusing again as this shows the actual OCEAN/DT values, so in this case the OCEAN there should be half:
> 
> Screen Shot 2022-03-30 at 10 30 28

Mmmm... Now I'm confused too 😅 Sorry if it sounds like a stupid question, but I haven't really gotten what this section should represent then. Can you please explain me why these values should be half as it not totally clear to me? If so, is this a math calc (can we just hardcoded) or this values should come from aquarius?

**Comment by mihaisc** [comment]
There is no math there, and it should be displayed. People need to know how many dts are in the pool (especially the owner). You get this data from the [subgraph ](https://v4.subgraph.rinkeby.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph/graphql?query=%7B%0A%20%20pool(id%3A%20%220xdf23f986e697645324593beab8269ac8aa62c3d2%22)%20%7B%0A%20%20%20%20id%0A%0A%20%20%20%20spotPrice%0A%20%20%20%20baseTokenLiquidity%0A%20%20%20%20datatokenLiquidity%0A%20%20%7D%0A%7D%0A)and you just display it

**Comment by mihaisc** [comment]
Indeed the base token pool liquidity is twice, not sure why. 
So a short recap for pool you get the stats directly from the subgraph , you only need to convert from pool shares to liquidity for the users/share owners. For them the trick now is how do we take in consideration the ss bot, this is why we decided to double the shares. The ss bot basically mirrors everyone else, so it will have 50% of the shares. To calculate the actual liquidity of an user we can just use the rule of three (we have total liquidity, total shares and user shares => user liqudity) . The problem with this is not exactly accurate so i'm more inclined to just call the contract `calcSingleOutGivenPoolIn`. We already do this in other places (add/remove liquidity and trade) so we have the mechanism in place.

**Comment by kremalicious** [comment]
> base token pool liquidity

yes, also figured this now with some subgraph queries where I remember some weeks ago this was correct where `baseTokenLiquidity` was the actual amount of base tokens, whereas now it seems this sends back the value of both tokens locked converted to baseToken.

**Comment by mihaisc** [comment]
So in this case the owner has 82.36 shares. simple rule of three results in 984.55 base token liqudity , if we double the share number then it's 2002.27 bt liquidity . If i call the contracts i get 1565.26 
image

**Comment by mihaisc** [comment]
@EnzoVezzaro you undid a good change, the first one was good. Don't stress on this issue until we have some better conclusions.

**Comment by EnzoVezzaro** [comment]
@mihaisc sorry about that. Restore initial PR with suggestion proposed by @kremalicious

File: src/@context/Pool/index.tsx [hunk 1]
```diff
@@ -119,11 +119,9 @@ function PoolProvider({ children }: { children: ReactNode }): ReactElement {
     if (!poolData) return
 
     // Total Liquidity
-    const totalLiquidityInOcean = isValidNumber(poolData.spotPrice)
-      ? new Decimal(poolData.baseTokenLiquidity).add(
-          new Decimal(poolData.datatokenLiquidity).mul(poolData.spotPrice)
-        )
-      : new Decimal(0)
+    const totalLiquidityInOcean = new Decimal(
+      poolData.baseTokenLiquidity ? poolData.baseTokenLiquidity : 0
```
[the remark this answers is not in the document]
Comment (EnzoVezzaro) [reply src/@context/Pool/index.tsx]: resolved

**Comment by kremalicious** [comment]
Those last numbers OCEAN/DT in pool stats are still wrong. There OCEAN should be half as it also shows the datatokens in this case.

Screen Shot 2022-03-31 at 14 03 18

File: src/@context/Pool/index.tsx [hunk 2]
```diff
@@ -119,11 +119,7 @@ function PoolProvider({ children }: { children: ReactNode }): ReactElement {
     if (!poolData) return
 
     // Total Liquidity
-    const totalLiquidityInOcean = isValidNumber(poolData.spotPrice)
-      ? new Decimal(poolData.baseTokenLiquidity).add(
-          new Decimal(poolData.datatokenLiquidity).mul(poolData.spotPrice)
-        )
-      : new Decimal(0)
+    const totalLiquidityInOcean = new Decimal(poolData.baseTokenLiquidity || 0)
```
Comment (kremalicious) [inline src/@context/Pool/index.tsx:122]: from UI standpoint this now makes sense but as said yesterday, based on this code it does not make sense. For whatever reason `poolData.baseTokenLiquidity` is already doubled, hence the issue with the OCEAN/DT further down

**Comment by mihaisc** [comment]
Nope, like we discussed, those are the only values that are ok. Those are the values from subgraph. Nothing to change there. The rest are wrong. So for user/owner liquidity we need to do a contract call (like we do when we remove liquidity)

**Comment by mihaisc** [comment]
To sumarize:
- fetch ocean liquidity for user and owner with `calcSingleOutGivenPoolIn` and then double it to represent value locked
- cleanup all the useless code
- this value needs to be doubled because it should be total tvl, basically `baseTokenLiquidity` * 2  ( it is coupled with this issue https://github.com/oceanprotocol/market/issues/1293) 
image
- this is fine as it is
image

**Comment by kremalicious** [comment]
Awesome, helps a lot. And yes, #1293 is meant only for UI development as all this `TokenList` madness is also v3 remnant and more complex than it should be. In here we continue only dealing with constructing the values

**Comment by EnzoVezzaro** [comment]
> 

Just to make I'm not losing pieces, did you see [this line](https://github.com/oceanprotocol/market/blob/fix/issue-1208-wrong-pool-tvl-value/src/@context/Pool/index.tsx#L150:L150)? In here we take into consideration the bot. According to this code the `Owner Value Locked` is already being multiplied by 2 so, in the example here is actually 984.573 OT (*2 for the bot) = 1,969.146 OT (closed to )

> To sumarize:
> 
> * fetch ocean liquidity for user and owner with `calcSingleOutGivenPoolIn` and then double it to represent value locked
> * cleanup all the useless code
> * this value needs to be doubled because it should be total tvl, basically `baseTokenLiquidity` * 2  ( it is coupled with this issue [Move pool TVL number #1293](https://github.com/oceanprotocol/market/issues/1293) , just for context, doesn't need to be handled in the same pr)
>   image
> * this is fine as it is
>   image

Applied changes:

- [X] fetch ocean liquidity for user and owner with calcSingleOutGivenPoolIn and then double it to represent value locked
- [X] cleanup all the useless code
- [X] this value needs to be doubled because it should be total tvl, basically baseTokenLiquidity * 2 ( it is coupled with this issue https://github.com/oceanprotocol/market/issues/1293 , just for context, doesn't need to be handled in the same pr)

**Comment by EnzoVezzaro** [comment]
> The values look ok . Merge main so we can test the tvl graph values. Those seem off. The values from [subgraph](https://v4.subgraph.rinkeby.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph/graphql?query=%7B%0A%20%20poolSnapshots(where%3A%7Bpool%3A%220xdf23f986e697645324593beab8269ac8aa62c3d2%22%7D)%0A%20%20%7B%0A%20%20%20%20baseTokenLiquidity%0A%20%20%7D%0A%7D) look good but the graph is not Image is from v4 branch image

Great, thanks. I'm still debugging an issue on Mumbai. The other networks seem to work fine.

Screen Shot 2022-03-31 at 11 43 50 AM

**Comment by mihaisc** [comment]
image
This is an rpc issue, check what id you have configured in local .env

**Comment by EnzoVezzaro** [comment]
> 

Might be that the problem them 🙌🏽 What values do i need on the .env file? I only have this variable set `NEXT_PUBLIC_INFURA_PROJECT_ID`. Do I need something else in the .env?

**Comment by EnzoVezzaro** [comment]
> well you need to make sure that the infura id is working for mumbai as well

Great! This totally solved my issue, I was debugging the connection itself this whole time. Thanks. Working on Mumbay now 👍🏽

**Review by mihaisc — APPROVED** [review]

**Pull request merged**